otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/ama-metrics-targetallocator.yaml

```diff
@@ -0,0 +1,221 @@
+{{- if or (ne .Values.AzureMonitorMetrics.ArcExtension true) (eq .Values.AzureMonitorMetrics.ArcEnableOperator true) }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    component: ama-metrics-targetallocator
+    kubernetes.azure.com/managedby: aks
+  name: ama-metrics-targetallocator
+  namespace: kube-system
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      rsName: ama-metrics-targetallocator
+      kubernetes.azure.com/managedby: aks
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        rsName: ama-metrics-targetallocator
+        kubernetes.azure.com/managedby: aks
+    spec:
+      containers:
+      - name: targetallocator
```

Comment:
Test out HA scenario to make sure TA replicas split targets exactly the same way between 2 replicas - Need to test later , not in scope of this PR